proto/external_storage.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+// purposely leaving out delete. that is up to individual external
+// storage server operators to do on their end
+
+service ProxyStorage {
+
+    rpc GetModuleModfile (ModuleMetadata) returns (ModuleMod) {}
+    rpc GetModuleInfo (ModuleMetadata) returns (ModuleInfo) {}
+    rpc GetModuleZip (ModuleMetadata) returns (ModuleZip) {}
+    rpc SaveModule (ModuleData) returns (ModuleSaveResponse) {}
+    rpc ListVersions (ListVersionsRequest) returns (ListVersionsResponse) {}
+    rpc CheckModuleAndVersion(ModuleMetadata) returns (CheckModuleAndVersionResponse) {}
```

Comment:
just my 2cents, i think this was a design flaw from the good old days, a nicer way to keep the interface small is to just interpret a `NotFound`  error as "not exists" :)

Reply:
good point, updated